draft-ietf-privacypass-protocol.md

```diff
@@ -158,6 +158,25 @@ Issuer directory resources have the media type "application/json"
 and are located at the well-known location /.well-known/token-issuer-directory;
 see {{wkuri-reg}} for the registration information for this well-known URI.
 
+Issuers SHOULD use HTTP caching to permit caching of this resource
+{{!RFC5861}}. The cache lifetime depends on the Issuer's key rotation schedule.
+Regular rotation of token keys is recommended to minimize the risk of key
+compromise.
+
+Issuers can control cache lifetime with the Cache-Control header, as follows:
+
+~~~
+  Cache-Control: max-age=86400
+~~~
+
+Consumers of the Issuer directory resource SHOULD follow the usual HTTP caching
+{{!RFC9111}} semantics when processing this resource. Long cache lifetimes may
+result in use of stale Issuer configuration information, whereas short
+lifetimes may result in decreased performance. When use of an Issuer
+configuration results in token issuance failures, e.g., because the
```

Comment:
Who is detecting a failure due to stale config? The origin or the client?

Reply:
Either of them could presumably detect a failure, though in practice it's usually the client.

Reply:
I think it might be useful to clarify this; since the client also gets the key information from a challenge, it seems a bit odd that the caching would mess it up?

Reply:
Well if the origin has a stale cached copy of the key and gives it to the client in the challenge, the client's issuance request would fail. I suppose we can be more specific, but it just seems like more words than required here.